File: demo/bridge-demo-ios/BridgeDemoIOS/LocationModule.swift
//
//  LocationModule.swift
//  BridgeDemoIOS
//
//  Created by Viethai Pham on 2/7/19.
//  Copyright © 2019 Viet Hai Pham. All rights reserved.
//

import RxSwift

public final class LocationModule {
  public struct Coordinate: ResponseType {
    public static func random() -> Coordinate {
      let latitude = Double.random(in: (0...100))
      let longitude = Double.random(in: (0...100))
      return Coordinate(latitude: latitude, longitude: longitude)
    }
    
    private let latitude: Double
    private let longitude: Double
    
    public func toDictionary() -> [String : Any?] {
      return ["latitude": self.latitude, "longitude": self.longitude]
    }
  }
  
  public func observeLocationChange() -> Observable<Coordinate> {
    return Observable<Int64>
      .interval(1, scheduler: MainScheduler.instance)
      .map({_ in Coordinate.random()})
  }
}

File: demo/bridge-demo-adr/app/src/main/java/com/viethai/demo/LocationModule.kt
package com.viethai.demo

import io.reactivex.Flowable
import java.util.*
import java.util.concurrent.TimeUnit

/** Created by viethai.pham on 2019-07-02 */
class LocationModule {
  data class Coordinate(private val latitude: Double, private val longitude: Double) {
    companion object {
      private val randomizer = Random()

      fun random(): Coordinate {
        val latitude = this.randomizer.nextDouble()
        val longitude = this.randomizer.nextDouble()
        return Coordinate(latitude, longitude)
      }
    }
  }

  fun observeLocationStream(): Flowable<Coordinate> {
    return Flowable.interval(1000, TimeUnit.MILLISECONDS).map { Coordinate.random() }
  }
}

